### PR TITLE
feat: ops proxy tasks

### PR DESCRIPTION
**Pull request by brandonchuah**
closes
- https://github.com/gelatodigital/ops-interface/issues/269
- https://github.com/gelatodigital/ops-interface/issues/214

### CODE REVIEW

File: contracts/examples/withTreasury/Counter.sol [hunk 1]
```diff
@@ -2,16 +2,28 @@
...
+
+    modifier onlyWhitelisted() {
+        require(
+            whitelisted[msg.sender] ||
+                msg.sender == ops ||
+                msg.sender == owner(),
+            "Counter: Not whitelisted"
+        );
+        _;
+    }
```
[the remark this answers is not in the document]
Comment (brandonchuah) [reply contracts/examples/withTreasury/Counter.sol]: created separate example for proxy tasks `CounterWithWhitelist` in 8dc42dccb26d99812c77ee8d95cc8c0da3792ed0

File: contracts/examples/withoutTreasury/CounterWithoutTreasury.sol [hunk 2]
```diff
@@ -2,22 +2,31 @@
...
+
+    modifier onlyWhitelisted() {
+        require(
+            whitelisted[msg.sender] ||
+                msg.sender == ops ||
+                msg.sender == owner(),
+            "Counter: Not whitelisted"
+        );
+        _;
+    }
```
[the remark this answers is not in the document]
Comment (brandonchuah) [reply contracts/examples/withoutTreasury/CounterWithoutTreasury.sol]: `8dc42dccb26d99812c77ee8d95cc8c0da3792ed0`

File: contracts/vendor/proxy/opsProxy/OpsProxyFactory.sol [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    /// @dev track deployed proxies
+    mapping(address => bool) internal _proxies;
+
+    /// @dev track proxy of user
+    mapping(address => address) internal _proxyOf;
+
+    modifier onlyOneProxy(address _account) {
+        require(
+            _proxyOf[_account] == address(0),
+            "OpsProxyFactory: Owns proxy"
```
[the remark this answers is not in the document]
Comment (brandonchuah) [reply contracts/vendor/proxy/opsProxy/OpsProxyFactory.sol]: 8dc42dccb26d99812c77ee8d95cc8c0da3792ed0

File: test/Ops-proxyTask.test.ts [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+      targets.push(ops.address);
+      values.push(0);
+    }
+
+    await opsProxy.connect(user).batchExecuteCall(targets, datas, values);
+
+    const tasks = await ops.getTaskIdsByUser(userAddress);
+    expect(tasks.length).to.be.eql(0);
+  });
+
```
[the remark this answers is not in the document]
Comment (brandonchuah) [reply test/Ops-proxyTask.test.ts:257]: 8dc42dccb26d99812c77ee8d95cc8c0da3792ed0

File: contracts/examples/withTreasury/Counter.sol [hunk 5]
```diff
@@ -2,6 +2,7 @@
 pragma solidity ^0.8.0;
 
 import {OpsReady} from "../../vendor/gelato/OpsReady.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
```
Comment (goums) [inline contracts/examples/withTreasury/Counter.sol:5]: I think it's not used?

File: contracts/examples/withoutTreasury/CounterWithoutTreasury.sol [hunk 6]
```diff
@@ -2,10 +2,8 @@
 pragma solidity ^0.8.0;
 
 import {OpsReady} from "../../vendor/gelato/OpsReady.sol";
-
-interface IOps {
-    function getFeeDetails() external view returns (uint256, address);
-}
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
```
Comment (goums) [inline contracts/examples/withoutTreasury/CounterWithoutTreasury.sol:5]: same, I think it's not used anymore?

**Review by goums — APPROVED** [review]

**Pull request closed without merging**